types/time.go

```diff
@@ -272,9 +273,19 @@ const (
 // NewTime constructs time from core time, type and fsp.
 func NewTime(coreTime CoreTime, tp uint8, fsp int8) Time {
 	t := ZeroTime
-	t.SetCoreTime(coreTime)
-	t.SetType(tp)
-	t.SetFsp(fsp)
+	p := (*uint64)(&t.coreTime)
```

Comment:
Please add comments for what will cause inlining not effective so that you used this way